src/scib_metrics/utils/_kmeans.py

```diff
@@ -120,6 +122,7 @@ def fit(self, X: np.ndarray):
         return self
 
     def _fit(self, X: np.ndarray):
+        self._pdists = pdist_squareform(X)
```

Comment:
this is potentially a big matrix. can we delete it after all the fits?

Reply:
Done